### PR TITLE
change base docker image from "python" to "jupyter/base-notebook"

### PR DESCRIPTION
**Pull request by chestercheng**
**SUMMARY**

I've changed the base docker image from "python" to "jupyter/base-notebook" for the compatible with infuseai.

@hlb Please review it. Is it correct?

### CODE REVIEW

File: Dockerfile [hunk 1]
```diff
@@ -1,35 +1,16 @@
...
 
 # Intstall jupyter extensions
-RUN jupyter contrib nbextension install --system
-RUN jupyter nbextension enable runtools/main
+USER root
+RUN pip install 'matplotlib==3.0.0' && \
+    pip install 'numpy==1.15.3' && \
+    pip install 'jupyter-contrib-nbextensions==0.5.0'
+RUN jupyter contrib nbextension install --system && \
+    jupyter nbextension enable runtools/main
```
Comment (yungyuc) [inline Dockerfile]: Love it.  Now I can `alt-f` and get all the cells run!

File: Dockerfile [hunk 2]
```diff
@@ -1,35 +1,16 @@
...
 MAINTAINER "chestercheng <chester7864@gmail.com>"
 
-# Install requirements
-RUN pip install matplotlib==3.0.0
-RUN pip install numpy==1.15.3
-RUN pip install ipykernel==5.1.0
-RUN pip install ipython==7.0.1
-RUN pip install jupyter==1.0.0
-RUN pip install jupyter-contrib-nbextensions==0.5.0
+ADD * /home/$NB_USER/work/
```
Comment (hlb) [inline Dockerfile]: @chestercheng I think we don't need $NB_USER too. We will use jovyan for people in all docker container :)

Comment (chestercheng) [reply Dockerfile]: @hlb ` $NB_USER ` is define in [jupyter/base-notebook Dockerfile](
https://github.com/jupyter/docker-stacks/blob/master/base-notebook/Dockerfile).
> Line 10.
> ```
> ARG NB_USER="jovyan"
> ...
> ```

It needs root permission when installing jupyter notebook extensions.
` $NB_USER ` at the last is make sure to switch back to non-root user (jovyan).

Comment (hlb) [reply Dockerfile]: oh great :D

**Comment by yungyuc** [comment]
@hlb let me know if the PR looks good to you and I’ll merge.

**Comment by hlb** [comment]
It works but runtools does not show in jupyterlab.

**Comment by yungyuc** [comment]
runtool works in my jupyter installation (OS X, outside docker).

@chestercheng could you help troubleshoot @hlb ‘s issue?

**Comment by chestercheng** [comment]
The docker image works well for me on OS X and Ubuntu.
@hlb may I have an InfuseAI account for test with this?

**Comment by chestercheng** [comment]
@yungyuc & @hlb
The runtools extension is not supported in JupyterLab. (ref: https://github.com/ipython-contrib/jupyter_contrib_nbextensions/issues/1048).
But we can use it in classic notebook.
screenshot 2018-11-08 at 3 14 24 pm

**Comment by yungyuc** [comment]
I can live with that and it agrees with my installation.  I tested it in Jupiter notebook, not lab.  @hlb is that OK?

**Comment by hlb** [comment]
@yungyuc sure :)

**Pull request merged**